pkg/engine/cas_template_engine.go

```diff
@@ -117,7 +118,7 @@ func initValues() (v map[string]interface{}) {
 		// list items is set as a top level property
 		string(v1alpha1.ListItemsTLP): map[string]interface{}{},
 		// task result is set as a top level property
-		string(v1alpha1.TaskResultTLP): nil,
+		string(v1alpha1.TaskResultTLP): map[string]interface{}{},
```

Comment:
The top level property TaskResult is initialized post execution of first run task. Since i am saving result before 'post' of first rt I need a non nil map.